### PR TITLE
Suggestion to create a pypi package for this

### PR DESCRIPTION
**Pull request by franklingu**
for #11

### CODE REVIEW

**Comment by franklingu** [comment]
```
(.venv) jgu@jgu-pc:~/self_repos/shell-functools (master%=) % pip uninstall ft
Uninstalling ft-1.0.0:
  /home/jgu/self_repos/shell-functools/.venv/bin/filter
  /home/jgu/self_repos/shell-functools/.venv/bin/foldl
  /home/jgu/self_repos/shell-functools/.venv/bin/ft-functions
  /home/jgu/self_repos/shell-functools/.venv/bin/map
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft-1.0.0-py3.6.egg-info
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__init__.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/__init__.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/command.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/error.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/functions.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/internal.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/termcolor.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/test_command.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/__pycache__/types.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/command.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/__init__.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/__pycache__/__init__.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/__pycache__/filter.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/__pycache__/foldl.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/__pycache__/map.cpython-36.pyc
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/filter.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/foldl.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/commands/map.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/error.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/functions.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/internal.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/termcolor.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/test_command.py
  /home/jgu/self_repos/shell-functools/.venv/lib/python3.6/site-packages/ft/types.py
Proceed (y/n)? y
  Successfully uninstalled ft-1.0.0
(.venv) jgu@jgu-pc:~/self_repos/shell-functools (master%=) % pip install .   
Processing /home/jgu/self_repos/shell-functools
Installing collected packages: ft
  Running setup.py install for ft ... done
Successfully installed ft-1.0.0
```

Example of running pip install .

**Comment by franklingu** [comment]
Issues to think about:
how to do versioning? maybe use ft.__version__ as the version in setup.py
what versions of python to support? I suggest this repo should have ci from travis -- I can make a separate PR for this one and verify builds across multiple python versions. I tested python 3.6 and 3.5. Not sure about others
more metadata to be added in setup? I am not sure about this

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, print_function
+import os
+
+from setuptools import setup, find_packages
+
+NAME = 'ft'
```
Comment (sharkdp) [inline setup.py]: `ft` is nice and short - but it's also hard to find. I'd prefer to call it `shell-functools` for now (unless someone has a better suggestion for a name)

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+VERSION = '1.0.0'
```
Comment (sharkdp) [inline setup.py]: I'd prefer to start with `0.2` because there might be some breaking changes until we hit `1.0`.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+PACKAGES = find_packages(
+    'ft', exclude=['*.tests', '*.tests.*', 'tests.*', 'tests']
+)
+PACKAGE_DIR = {
+    '': 'ft'
+}
+PACKAGE_DATA = {
+    '': ['']
+}
+AUTHOR = 'sharkdp'
```
Comment (sharkdp) [inline setup.py]: You can put `David Peter` here.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+AUTHOR_EMAIL = ''
```
Comment (sharkdp) [inline setup.py]: You can put `mail@david-peter.de` here.

**Comment by sharkdp** [comment]
> how to do versioning? maybe use ft.version as the version in setup.py

How do other pip packages handle this? I'm okay with adding a new python file somewhere that includes the `__version__` (or include it in an existing file).

> what versions of python to support? I suggest this repo should have ci from travis -- I can make a separate PR for this one and verify builds across multiple python versions.

That sounds great.

> I tested python 3.6 and 3.5. Not sure about others

I'd be okay with that for now.

> more metadata to be added in setup? I am not sure about this

I can take a look once this is merged.

File: setup.py (hunk 1 above)
Comment (franklingu) [reply setup.py]: Both names are not taken. https://pypi.python.org/pypi?%3Aaction=search&term=shell-functools&submit=search. I am okay with either personally

File: setup.py (hunk 2 above)
Comment (franklingu) [reply setup.py]: OK. will move this to ft.__version__

**Comment by sharkdp** [comment]
Thanks!

**Comment by franklingu** [comment]
maybe you can create tags now for this repo

**Comment by franklingu** [comment]
https://packaging.python.org/tutorials/distributing-packages/#uploading-your-project-to-pypi

you can take a look at this and upload the package to pypi and publish it!

**Comment by sharkdp** [comment]
Thank you for the link.

The package has been uploaded to pypi.

**Pull request merged**